### PR TITLE
[Doc] Describe the difference between ANN and KNN, and how to create indices.

### PR DESCRIPTION
**Pull request by eddyxu**

### CODE REVIEW

**Comment by eddyxu** [comment]
Added a FAQ section.

File: docs/src/ann_indexes.md [hunk 1]
```diff
@@ -138,3 +152,31 @@ You can select the columns returned by the query using a select clause.
...
+Product quantization can lead to approximately `16 * sizeof(float32) / 1 = 64` times of space reduction.
+
+### How to choose `num_partitions` and `num_sub_vectors` for `IVF_PQ` index.
+
+`num_partitions` is used to decide how many partitions the first level `IVF` index uses.
+Higher number of partitions could lead to more efficient I/O during queries and better accuracy, but it takes much more time to train.
+On `SIFT-1M` dataset, our benchmark shows that keeping each partition 1K-4K rows lead to a good latency / recall.
+
+`num_sub_vectors` decides how many Product Quantization code to generate on each vector. Because
+Product Quantization is a loss compression of the original vector, the more `num_sub_vectors` usually results to
```
Comment (changhiskhan) [inline docs/src/ann_indexes.md:180]: Lossy

File: docs/src/ann_indexes.md [hunk 2]
```diff
@@ -138,3 +152,31 @@ You can select the columns returned by the query using a select clause.
...
+For example, with a 1024-dimension dataset, if we choose `num_sub_vectors=64`, each sub-vector has `1024 / 64 = 16` float32 numbers.
+Product quantization can lead to approximately `16 * sizeof(float32) / 1 = 64` times of space reduction.
+
+### How to choose `num_partitions` and `num_sub_vectors` for `IVF_PQ` index.
+
+`num_partitions` is used to decide how many partitions the first level `IVF` index uses.
+Higher number of partitions could lead to more efficient I/O during queries and better accuracy, but it takes much more time to train.
+On `SIFT-1M` dataset, our benchmark shows that keeping each partition 1K-4K rows lead to a good latency / recall.
+
+`num_sub_vectors` decides how many Product Quantization code to generate on each vector. Because
```
Comment (changhiskhan) [inline docs/src/ann_indexes.md:179]: Maybe we should just add an explicit check that num_subvectors < dimensions or else raise an exception?

**Review by changhiskhan — APPROVED** [review]
Two nits

**Review by AyushExel — APPROVED** [review]
Thanks! looks great. I think there are couple comments by Chang, other than that looks good to go.

File: docs/src/ann_indexes.md (hunk 2 above)
Comment (eddyxu) [reply docs/src/ann_indexes.md:179]: sure can do.  I've seen other db uses a `dim ^ 0.5` as default value too.

File: docs/src/ann_indexes.md (hunk 1 above)
Comment (eddyxu) [reply docs/src/ann_indexes.md:180]: Fixed.

**Pull request merged**